conanfile.py

```diff
@@ -322,6 +322,7 @@ def requirements(self):
                 or self.options.build_rsa_discovery_common or self.options.build_rsa_remote_service_admin_dfi
                 or self.options.build_launcher):
             self.requires("libcurl/[>=8.4.0 <9.0.0]")
+            self.requires("openssl/[>=3.2.0]")
```

Comment:
This change was needed to prevent build failures due to the dynamic linking error: `version OPENSSL_3.2.0 not found.` See, for example: https://github.com/apache/celix/actions/runs/9538818641/job/26288458779

I also noted that when running ctest after sourcing conanrun.sh, you get the following message:
`ctest: /root/.conan2/p/b/libcu05433fa1f2d4d/p/lib/libcurl.so.4: no version information available (required by ctest).`

When using ctest as follows, this message disappears:
`ctest --output-on-failure --test-command ./workspaces/celix/build/conanrun.sh`

Reply:
We don't depend on openssl directly, and thus it is better to use `self.requires("openssl/[>=3.2.0]", override=True)` to avoid introducing an extra direct dependency.